packages/eslint-plugin-wpcalypso/lib/rules/no-package-relative-imports.js

```diff
@@ -0,0 +1,111 @@
+const fs = require( 'fs' );
+const path = require( 'path' );
+
+const getRelativeImports = ( relativeDir ) => {
+	return fs.readdirSync( relativeDir, { withFileTypes: true } ).map( ( file ) => {
+		const fileName = file.name;
+		if ( file.isDirectory() ) return fileName;
+		const extension = path.extname( fileName );
+		if ( extension !== '.js' ) return fileName;
+		return path.basename( fileName, extension );
+	} );
+};
+
+const reportIfPackageRelative = ( { context, node, dependency } ) => {
+	const { mapping, warnOnDynamicImport } = context.options[ 0 ];
```

Comment:
These options don't appear to be documented. That's not a blocking concern for this PR, considering the early state of this package, but something to consider.